ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java

```diff
@@ -118,15 +122,29 @@ public void go(final AcceptsOneWidget container) {
     }
 
     private void loadImporters() {
-        final Map<String, Set<ProjectImporterDescriptor>> importersByCategory = new HashMap<>();
+        final Map<String, Set<ProjectImporterDescriptor>> importersByCategory = new LinkedHashMap<>();
 
-        final Unmarshallable<List<ProjectImporterDescriptor>> unmarshaller =
-                dtoUnmarshallerFactory.newListUnmarshaller(ProjectImporterDescriptor.class);
+        final Unmarshallable<ProjectImporterData> unmarshaller =
+                dtoUnmarshallerFactory.newUnmarshaller(ProjectImporterData.class);
 
-        AsyncRequestCallback<List<ProjectImporterDescriptor>> callback =
-                new AsyncRequestCallback<List<ProjectImporterDescriptor>>(unmarshaller) {
+        AsyncRequestCallback<ProjectImporterData> callback =
+                new AsyncRequestCallback<ProjectImporterData>(unmarshaller) {
                     @Override
-                    protected void onSuccess(List<ProjectImporterDescriptor> result) {
+                    protected void onSuccess(ProjectImporterData data) {
+                        List<ProjectImporterDescriptor> result = data.getImporters();
+                        String defaultImporterId = data.getConfiguration().get(DEFAULT_PROJECT_IMPORTER);
+                        Iterator<ProjectImporterDescriptor> itr = result.iterator();
```

Comment:
first we need to find default importer and put it and it's category into result map at first position, then we'll handle the rest of importers.